bff/src/Altinn.Authentication.UI/Altinn.Authentication.UI.Core/SystemUsers/SystemUserService.cs

```diff
@@ -50,13 +54,60 @@ public async Task<List<SystemUser>> GetAllSystemUserDTOsForChosenUser(int id, Ca
 
     public async Task<SystemUser?> PostNewSystemUserDescriptor(int partyId, SystemUserDescriptor newSystemUserDescriptor, CancellationToken cancellation = default)
     {
-        AuthorizedPartyExternal party = await _partyLookUpClient.GetPartyFromReporteeListIfExists(partyId);
-        string partyOrgNo = party.OrganizationNumber; 
-        return await _systemUserClient.PostNewSystemUserReal(partyOrgNo, newSystemUserDescriptor, cancellation);
+        
+
+        AuthorizedPartyExternal party = await _partyLookUpClient.GetPartyFromReporteeListIfExists(partyId);        
+        string partyOrgNo = party.OrganizationNumber;
+        bool canDelegate = await UserDelegationCheckForReportee(partyId, newSystemUserDescriptor.SelectedSystemType!, cancellation);
+        if (!canDelegate) return null;
+        
+        SystemUser? systemUser; systemUser = await _systemUserClient.PostNewSystemUserReal(partyOrgNo, newSystemUserDescriptor, cancellation);
+        if(systemUser is null) return null;
+
+        List<Right> rights = await _systemRegisterClient.GetRightFromSystem(systemUser.SystemId , cancellation);
+
+        bool delagationSucceeded = await _partyLookUpClient.DelegateRightToSystemUser(partyId.ToString(),systemUser, rights);
+        if (delagationSucceeded) return systemUser;
+
+        return null;        
     }
 
     public async Task<bool> ChangeSystemUserProduct(string selectedSystemType, Guid id, CancellationToken cancellationToken = default)
     {
         return await _systemUserClient.ChangeSystemUserRealProduct(selectedSystemType, id, cancellationToken);
     }
+
+    public async Task<bool> UserDelegationCheckForReportee(int partyId, string systemId ,CancellationToken cancellationToken = default)
```

Comment:
We need to use the response from CheckDeleagtionAccess for delegation

Reply:
ok implemented